### PR TITLE
feat(rich-text-editor): move toolbar plugin groups to portal

### PR DESCRIPTION
**Pull request by augustobmoura**
[FX-4137]

### Description

On this PR, we are wrapping all toolbar plugins on groups. For now, every Toolbar has its group, in the future, we can think of a mechanism for controlling groups more precisely

### How to test

- Temploy
- Check happo and temploy, last toolbar vertical bar

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| image | image |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [ ] Covered with tests

**Breaking change**

- [NA] codemod is created and showcased in the changeset
- [NA] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4137]: https://toptal-core.atlassian.net/browse/FX-4137?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by augustobmoura** [comment]
@toptal-anvil ping reviewers

File: packages/picasso-rich-text-editor/src/RichTextEditorToolbar/RichTextEditorToolbar.tsx [hunk 1]
```diff
@@ -71,13 +72,19 @@ export const RichTextEditorToolbar = forwardRef<HTMLDivElement, Props>(
...
     const allowEmojis = plugins?.includes('emoji')
 
     return (
-      <Container id={`${id}toolbar`} ref={ref} className={classes.toolbar}>
+      <Container
+        id={`${id}toolbar`}
+        ref={toolbarRef}
+        className={classes.toolbar}
+      >
         <Container
```
Comment (dmaklygin) [inline packages/picasso-rich-text-editor/src/RichTextEditorToolbar/RichTextEditorToolbar.tsx:88]: why not to set the ref to this container?

File: packages/picasso/src/utils/use-multiple-forward-refs.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ *   console.log(ref2.current) // <div />
+ */
+const useMultipleForwardRefs = <T>(refs: ForwardedRef<T>[]) =>
+  useCallback(
+    (refValue: T) => {
+      for (const ref of refs) {
+        forwardRef(ref, refValue)
+      }
+    },
+    [...refs]
```
Comment (dmaklygin) [inline packages/picasso/src/utils/use-multiple-forward-refs.ts:34]: this dependency is kinda dangerous. `refs` may not be changed, however, `ref.current` was changed.

Comment (augustobmoura) [reply packages/picasso/src/utils/use-multiple-forward-refs.ts:34]: `ref.current` changes shouldn't trigger re-renders, we are only interested in the ref holders themselves changing.

File: packages/picasso-rich-text-editor/src/RichTextEditorToolbar/RichTextEditorToolbar.tsx (hunk 1 above)
Comment (augustobmoura) [reply packages/picasso-rich-text-editor/src/RichTextEditorToolbar/RichTextEditorToolbar.tsx:88]: wdym? `ref` is being set
On line 75 we are sharing the ref with both the Toolbar portal and with the parent component

File: packages/picasso-rich-text-editor/src/plugins/Toolbar/Toolbar.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    <>
+      {element &&
+        createPortal(
+          <Container className={classes.group}>{children}</Container>,
+          element,
+          keyName
+        )}
+    </>
+  )
+}
```
[the remark this answers is not in the document]
Comment (augustobmoura) [reply packages/picasso-rich-text-editor/src/plugins/Toolbar/Toolbar.tsx:63]: There are a few inconsistencies with the suggestion and some unnecessary code, but I think we have simplify the original code with the proposed `if` guard

File: packages/picasso/src/utils/use-multiple-forward-refs.ts (hunk 2 above)
Comment (augustobmoura) [reply packages/picasso/src/utils/use-multiple-forward-refs.ts:34]: We need to make sure that the elements on the array are also checked against useCallback as we are looping through them on `for (const ref of refs) {`

Comment (augustobmoura) [reply packages/picasso/src/utils/use-multiple-forward-refs.ts:34]: The other function looks similar, but it has some weird behavior. It doesn't keep two refs synced. It only syncs them on re-render. I'm not sure if that is the intended implementation or a mistake, I would keep it the old way so we don't break anything

**Review by TomasSlama — APPROVED** [review]

File: packages/picasso-rich-text-editor/src/RichTextEditorToolbar/RichTextEditorToolbar.tsx (hunk 1 above)
Comment (dmaklygin) [reply packages/picasso-rich-text-editor/src/RichTextEditorToolbar/RichTextEditorToolbar.tsx:88]: What's the point to sharing ref if we can assign the second one on Container (line: 81)

**Review by dmaklygin — APPROVED** [review]

**Pull request merged**